### ISSUE TITLE
Add Sentry for the Swift codebase

### ISSUE THREAD

**Issue by jamilbk** [CloudFire-LLC/cloudfire#7560]

**Comment by thomaseizinger** [comment]
It would be helpful to also add an FFI function to set the account-slug in the Rust telemetry context while we are at it.

**Comment by jamilbk** [comment]
> It would be helpful to also add an FFI function to set the account-slug in the Rust telemetry context while we are at it.

Hm yeah this got me thinking. Ideally, wouldn't we just want one Sentry context active?

To achieve that on Apple I guess we'd have to keep Sentry in the Swift side and then expose telemetry callbacks for the Rust code to call right?

On the Rust side, you'd have a module that either calls the Swift/Kotlin telemetry functions or calls another Rust-based module, similar to how we do today with the connlib callbacks.

The Swift SDK is quite nice and ties deeply into macOS/iOS and even the SwiftUI view layer if we want.

**Comment by thomaseizinger** [comment]
> > It would be helpful to also add an FFI function to set the account-slug in the Rust telemetry context while we are at it.
> 
> Hm yeah this got me thinking. Ideally, wouldn't we just want one Sentry context active?

I don't think it matters much if we have two Sentry contexts in different language runtimes active. They don't share any state so they shouldn't interfere with each other.

**Comment by jamilbk** [comment]
> > > It would be helpful to also add an FFI function to set the account-slug in the Rust telemetry context while we are at it.
> > 
> > 
> > Hm yeah this got me thinking. Ideally, wouldn't we just want one Sentry context active?
> 
> I don't think it matters much if we have two Sentry contexts in different language runtimes active. They don't share any state so they shouldn't interfere with each other.

K. Just need to remember to keep the iOS network extension process memory usage under 40ish MB. I'll do a cursory check to see what we're at now.

**Comment by thomaseizinger** [comment]
From memory, the headless client uses about 13MB right now.
